### ISSUE TITLE
Support more metrics in PHE

### ISSUE THREAD

**Issue by LeoGrin** [PriorLabs/tabpfn-extensions#13]
Some discussion in #12 
Some points to consider: maybe for some metrics we want to use `balance_probabilities` or add it to the search space. How well do we do on the available metrics? We also probably want to pass the metric the the TabPFN model itself.

**Comment by LennartPurucker** [comment]
FYI: for supporting almost everything sklearn supports, I can recommend using [AutoGluon["core"]](https://github.com/autogluon/autogluon/tree/master/core/src/autogluon/core/metrics) to get its metric handling or [vendoring it like I did for fine-tuning](https://github.com/LennartPurucker/finetune_tabpfn/tree/master/src/finer/metric_utils).
